opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -235,6 +244,71 @@ public List<DateFormatter> getAllCustomFormatters() {
         .collect(Collectors.toList());
   }
 
+  /**
+   * Retrieves a list of custom formatters and OpenSearch named formatters defined by the user, and
+   * attempts to parse the given date/time string using these formatters.
+   *
+   * @param dateTime The date/time string to parse.
+   * @return A ZonedDateTime representing the parsed date/time in UTC, or null if parsing fails.
+   */
+  public ZonedDateTime getParsedDateTime(String dateTime) {
+    List<DateFormatter> dateFormatters = this.getAllNamedFormatters();
+    dateFormatters.addAll(this.getAllCustomFormatters());
+    ZonedDateTime zonedDateTime = null;
+
+    // check if dateFormatters are empty, then set default ones
+    if (dateFormatters.isEmpty()) {
+      dateFormatters = initializeDateFormatters();
+    }
```

Comment:
```
List<DateFormatter> dateFormatters = this.getAllNamedFormatters();
dateFormatters.addAll(this.getAllCustomFormatters());
```
this code will duplicate getAllCustomFormatters?

use Stream.concat to concat two list works?

Reply:
formats is fetched from OpenSearch index mapping, if formats is null, does it means the OpenSearchDataType is  a wrapper of CoreType. So in this case, should we use OPENSEARCH_DEFAULT_FORMATS to format to parse the data?

Reply:
When format is null, it can still be `OpenSearchDateType` field as there can be date fields without formats (default ones) defined. So I have used OPENSEARCH_DEFAULT_FORMATS to format the date object in those cases.